src/Test/Utils/base-setup.tsx

```diff
@@ -1,71 +1,31 @@
 import React from "react";
 import { QueryClientProvider } from "@tanstack/react-query";
 import { StoreProvider } from "easy-peasy";
-import { RemoteData } from "@/Core";
-import {
-  QueryResolverImpl,
-  getStoreInstance,
-  CommandResolverImpl,
-  QueryManagerResolverImpl,
-  CommandManagerResolverImpl,
-} from "@/Data";
-import {
-  StaticScheduler,
-  DeferredApiHelper,
-  dependencies,
-  EnvironmentDetails,
-  EnvironmentSettings,
-} from "@/Test";
-import { UrlManagerImpl } from "@/UI";
+import { getStoreInstance } from "@/Data";
+import { StaticScheduler, DeferredApiHelper, MockedDependencyProvider } from "@/Test";
 import { BlockingModal } from "@/UI/Components";
-import { DependencyProvider } from "@/UI/Dependency";
-import { PrimaryRouteManager } from "@/UI/Routing";
+import * as envModifier from "@/UI/Dependency/EnvironmentModifier";
 import { TestMemoryRouter } from "@/UI/Routing/TestMemoryRouter";
 import { testClient } from "./react-query-setup";
 
 export function baseSetup(Page: React.ReactNode, halted: boolean = false) {
+  jest.spyOn(envModifier, "useEnvironmentModifierImpl").mockReturnValue({
```

Comment:
this will be removed at some point after the refactoring?

Reply:
mocking it like this, could hide some bugs over time, so if we can remove it in a later stage, I think it would be safer

Reply:
but how does that differ compared to mocking like we used to? 

I'm totally open to change that but I guess that if we  have modifier tested then here is no room for hidden bugs as we use an actual envModifier with only one function mocked.

Reply:
I'm probably missing some pieces, but for me, the "new" implementation should only require mocking external libraries like jointJs, or monaco, and the api responses with MSW to build the page for the test setup. 
The page/test suit we have, that use the full page as context, are more integration tests than unit tests. And yes, the Modifier might be covered in a unit case, but it should work in a full integration too. This is why I'm making this comment, but we can review this in a later stage.

Reply:
okay, now I see your point of view more clearly. Makes totally sense.  I here firstly went with the flow as it was that some of these managers/handlers are mocked. then that parameter has to go and we need to add extra call for the environment to access halting setting. 
fine by me,I added that comment to the task about